File: LiveDataDemo/app/src/main/java/guandong/com/livedatademo/adapter/ItemAdapter.java
package guandong.com.livedatademo.adapter;

import android.support.annotation.Nullable;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;

import java.util.List;

import guandong.com.livedatademo.R;

/**
 * @author guandongchen
 * @date 2018/3/22
 */

public class ItemAdapter extends BaseQuickAdapter<String,BaseViewHolder>{
    public ItemAdapter(int layoutResId, @Nullable List<String> data) {
        super(layoutResId, data);
    }

    public ItemAdapter(@Nullable List<String> data) {
        super(data);
    }

    public ItemAdapter(int layoutResId) {
        super(layoutResId);
    }

    @Override
    protected void convert(BaseViewHolder helper, String item) {
        helper.setText(R.id.textView,item);
    }
}

File: LiveDataDemo/app/src/main/java/guandong/com/livedatademo/adapter/ItemAdapter.kt
package guandong.com.livedatademo.adapter

import com.chad.library.adapter.base.BaseQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder

import guandong.com.livedatademo.R

/**
 * @author guandongchen
 * @date 2018/3/22
 */

class ItemAdapter : BaseQuickAdapter<String, BaseViewHolder> {
    constructor(layoutResId: Int, data: List<String>?) : super(layoutResId, data) {}

    constructor(data: List<String>?) : super(data) {}

    constructor(layoutResId: Int) : super(layoutResId) {}

    override fun convert(helper: BaseViewHolder, item: String) {
        helper.setText(R.id.textView, item)
    }
}
